packages/medusa/src/services/middleware.ts

```diff
@@ -1,31 +1,51 @@
+import { RequestHandler, Router } from "express"
 import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+import { TransactionBaseService } from "../interfaces"
+
+type middlewareType = {
+  middleware: (options: Record<string, unknown>) => RequestHandler
+  options: Record<string, unknown>
+}
 
 /**
  * Orchestrates dynamic middleware registered through the Medusa Middleware API
  */
-class MiddlewareService {
-  constructor(container) {
+class MiddlewareService extends TransactionBaseService<MiddlewareService> {
```

Comment:
todo: shouldn't be necessary to convert this to a transactions service as it doesn't interact with the db layer

Reply:
indeed, this service does not need that extension.

Reply:
I was on the fence but included it for consistency, but I will remove it 😄